tests/api/api/test_tags.py

```diff
@@ -239,6 +239,51 @@ def test_append_artifact_tags_by_uid_identifier_latest(
         )
         assert response_body["artifacts"][0]["metadata"]["name"] == artifact1_name
 
+    def test_append_artifact_tags_with_charachters(
+        self, db: sqlalchemy.orm.Session, client: fastapi.testclient.TestClient
+    ):
+        """
+        This tests the following scenarios: adding an invalid tag to an existing project,
+        and attempting to create a project with an invalid tag. and ensure that both fail
+        """
+
+        self._create_project(client)
+        valid_tag_name = "valid-tag"
+        invalid_tag_name = "tag$%^#"
+        artifact1_labels = {"artifact_name": "artifact1"}
+        artifact2_labels = {"artifact_name": "artifact2"}
+
+        _, _, artifact1_name, artifact1_uid, artifact1_key, _, _ = self._store_artifact(
+            client,
+            tag=invalid_tag_name,
+            uid="latest",
+            labels=artifact1_labels,
+            expected_status_code=http.HTTPStatus.BAD_REQUEST.value,
+        )
+
+        _, _, artifact1_name, artifact1_uid, artifact1_key, _, _ = self._store_artifact(
+            client, tag=valid_tag_name, uid="latest", labels=artifact2_labels
+        )
+
+        response = self._append_artifact_tag(
+            client=client,
+            tag=invalid_tag_name,
+            identifiers=[
+                mlrun.api.schemas.ArtifactIdentifier(
+                    key=artifact1_key, uid=artifact1_uid
+                ),
```

Comment:
would be worth to test also `_overwrite_artifact_tags` to see that `invalid tag_name` fails, and `_delete_artifact_tag` does work to delete the invalid tag name.